internal/appsec/waf.go

```diff
@@ -492,3 +493,13 @@ func addWAFMonitoringTags(th tagsHolder, rulesVersion string, overallRuntimeNs,
 	th.AddTag(wafDurationTag, float64(internalRuntimeNs)/1e3)   // ns to us
 	th.AddTag(wafDurationExtTag, float64(overallRuntimeNs)/1e3) // ns to us
 }
+
+func addressesContains(addresses []string, address string) bool {
```

Comment:
Either move to `map[string]struct{}`, or use `strings.Search` please.
We were thinking of the first option for a while. Maybe it's a good time to do it, unless this is a big breaking change in the code base, in which case I suggest you postpone it.

Reply:
switched to `map[string]struct{}`, There was better refactoring to do but this will be on another scale I think

Reply:
resolving after @Hellzy re-review